pgrx/src/datum/array.rs

```diff
@@ -628,6 +623,11 @@ impl<'a, T: FromDatum> Iterator for ArrayIterator<'a, T> {
         let Some(is_null) = array.null_slice.get(*curr) else { return None };
         *curr += 1;
 
+        // Make sure a previous iter didn't push out of bounds!
+        #[cfg(debug_assertions)]
+        if !is_null {
```

Comment:
Nit: I'd prefer `if cfg!(debug_assertions) && !is_null` instead of the `#[cfg(...)]`, to avoid annoyances from conditional compilation (future changes missing a rename or having unused var warnings -- altho they'd be caught quickly).

Or even `if !is_null { debug_assert!(...); }`

Reply:
clearly it should be
```suggestion
        // If we might read, make sure a previous iter didn't push out of bounds!
        debug_assert!(is_null || *ptr < array.raw.end_ptr());
```